doc/contribute/coding_guidelines/index.rst

```diff
@@ -1297,6 +1297,101 @@ macros. Given that Zephyr uses a fork of the corresponding upstream for each
 module, it is always possible to patch the macro implementation in each module
 to avoid collisions.
 
+.. _coding_guideline_libc_usage_restrictions_in_zephyr_kernel:
+
+Rule A.4: C Standard Library Usage Restrictions in Zephyr Kernel
+================================================================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+Zephyr kernel shall not use the C standard library functions and macros that
+are not available in the :ref:`minimal libc <c_library_minimal>`.
+
+The "Zephyr kernel" in this context consists of the following components:
+
+* Kernel (:file:`kernel`)
+* OS Library (:file:`lib/os`)
+* Architecture Port (:file:`arch`)
+* Logging Subsystem (:file:`subsys/logging`)
+
+The full list of the C standard library functions that are available as part of
+the minimal libc can be found :ref:`here <c_library_minimal_functions>`.
+
+Rationale
+---------
+
+Zephyr kernel must be able to build with the
+:ref:`minimal libc <c_library_minimal>`, a limited C standard library
+implementation that is part of the Zephyr RTOS and maintained by the Zephyr
+Project, to allow self-contained testing and verification of the kernel and
+core OS services.
+
+In order to ensure that the Zephyr kernel can build with the minimal libc, it
+is necessary to restrict the use of the C standard library functions and macros
+in the Zephyr kernel to the functions and macros that are available as part of
+the minimal libc.
+
+Rule A.5: C Standard Library Usage Restrictions in Zephyr Codebase
+==================================================================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+Zephyr codebase shall not use the functions and macros that are not defined by
+the ISO C standard, unless exempted by this rule.
+
+The "Zephyr codebase" in this context refers to all source code files committed
+to the `main Zephyr repository`_, except the Zephyr kernel as defined by the
+:ref:`coding_guideline_libc_usage_restrictions_in_zephyr_kernel`.
+
+The following non-ISO functions and macros are exempt from this rule and may be
```

Comment:
I think this is somewhat awkwardly specified. I think we should define the Zephyr C library API as being a combination of the ISO/IEC standard C library along with a few functions borrowed from other standards.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/57598#discussion_r1190272409

I will update the Rule A.5 to explicitly refer to the ISO/IEC 9899.

Reply:
Updated the coding guidelines to refer to to the ISO/IEC 9899:2011.